_posts/2016-12-08-underhanded-rust.en.md

```diff
@@ -0,0 +1,216 @@
+expression like `libc = "= 0.2.17"`.
+
+* You are also welcome to simulate introducing bugs into your dependencies. To
+  protect our ecosystem, please do not actually submit patches upstream, but
+instead land your patches in a fork of the upstream project and depend on them
+with
+[git](http://doc.crates.io/specifying-dependencies.html#specifying-dependencies-from-git-repositories)
+or
+[path](http://doc.crates.io/specifying-dependencies.html#specifying-path-dependencies)
+dependencies. These patches will then be reviewed and incorporated into the
+scoring of your submission.
+
+* Exploits based on human perception, like mistaking an `l` for a `1`, are
+  worth just as much as "hard" errors. The goal is a clever vulnerability that
+  passes visual inspection, whatever the mechanics of the underlying bug.
+
+* Underhandedness which can be plausibly explained as an innocent programming
+  error is worth more points.
+
+* Submissions are worth more points if you implement your solution without
+  using unsafe blocks. However, clever use of unsafe blocks which are resilient
+  to the high level of scrutiny typically expected of unsafe code can be worth
+  bonus points too.
+
+* Extra points are awarded for code which includes and passes its own tests.
+  Additionally, extra points if the underhandedness is not revealed by the
+  rustc or [clippy](https://github.com/Manishearth/rust-clippy) lints.
+
+* Extra points are awarded for creativity and funny bugs.
+
+# Submission Guidelines and Deadlines
+
+Send your submissions to <mailto:underhanded@rust-lang.org> by March 1, 2017.
+
+To make things easier for us to judge, we require you to send us your
+submission in the following format. Please send your submission as a compressed
+folder (`.tar.gz`, `.tar.bz2`, `.zip`, etc) with the following contents:
+
+* `README` - an explanation of how to run your submission and verify the
+  exploit worked without revealing your exploit technique.
+
+* `README-EXPLOIT` - an explanation of how your exploit works and why it’s hard
+  to detect.
+
+* `AUTHORS` - the list of people who worked on your entry.
+
+* `LICENSE` - The open source license your submission is released under (CC0,
+  GPL, MIT, BSD, Apache, etc). Your submission MUST include a license.
+
+* `submission/` - A directory containing the technical contents of your
+  submission.
+
+* `blogpost/` - A directory containing a "blog post style" explanation of your
+  entry. Please put the blog post contents in a
+  [Markdown](https://daringfireball.net/projects/markdown/) file. Please include
+  images in this directory if it will help explain your entry. It may be
+  necessary to give a higher level explanation than you did in README-EXPLOIT as
+  your reader might not be as experienced as the judges. Should you struggle with
+  writing prose in english, please submit this in your favourite language and we
+  will assist in translation.
```

Comment:
yeah, I think I'll replace this with "you may be asked to write a blogpost about your entry" in order to cut out some of the work for the submitters.